.github/workflows/ci.yml

```diff
@@ -16,25 +16,42 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  get_version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
+
+    steps:
+      - name: Check out pipeline code
+        uses: actions/checkout@v3
+
+      - name: Get Minimum supported Nextflow version
+        id: get_version
+        run: |
+          VERSION=$(grep 'nextflowVersion' nextflow.config | awk -F'= ' '{print $2}' | tr -d '!>=" ')
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
   test:
-    name: Run ${{ matrix.aligner }}-${{ matrix.profile }}-${{ matrix.NXF_VER }}
-    # Only run on push if this is the nf-core dev branch (merged PRs)
-    if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/methylseq') }}"
+    name: Run ${{ matrix.profile }} pipeline test
+    needs: get_version
+
+    if: "${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/fetchngs') }}"
+
     runs-on: ubuntu-latest
+    env:
+      NXF_VER: ${{ needs.get_version.outputs.version }}
+
     strategy:
       matrix:
-        NXF_VER:
-          - "22.10.1"
-          - "latest-everything"
-        aligner:
-          - bismark
-          - bismark_hisat
-          - bwameth
-        profile: ["docker"] # TODO , "singularity", "conda"]
```

Comment:
We need some form of a matrix here, or the tests are going to take hours.

Reply:
Or to implement the check for changes, but I'd rather follow up with a seperate PR.